src/unix/mmap.c

```diff
@@ -224,6 +224,10 @@ sysreturn mremap(void *old_address, u64 old_size, u64 new_size, int flags, void
 
 static sysreturn mincore(void *addr, u64 length, u8 *vec)
 {
+    u64 where = u64_from_pointer(addr);
+    if (where & MASK(PAGELOG))
+        return -EINVAL;
+
     if (validate_virtual(addr, length)) {
```

Comment:
Could we also pad the length up here? e.g. length = pad(length, PAGESIZE);

So actually, this implementation of mincore has long bitrotted and isn't really sufficient. I just filed an issue for this - #956.